src/psyclone/dynamo0p3_builtins.py

```diff
@@ -1016,55 +1047,226 @@ def gen_code(self, parent):
         :type parent: :py:class:`psyclone.f2pygen.BaseGen`
 
         '''
-        # Sum all the elements of a field. The variable holding the
-        # sum is initialised to zero in the psy layer.
+        # Sum all the elements of a real-valued field. The real scalar
+        # variable holding the sum is initialised to zero in the PSy layer.
         field_name = self.array_ref(self._arguments.args[1].proxy_name)
         sum_name = self._reduction_ref(self._arguments.args[0].name)
         rhs_expr = sum_name + "+" + field_name
         parent.add(AssignGen(parent, lhs=sum_name, rhs=rhs_expr))
 
 
+# ******************************************************************* #
+# ************** Built-ins for integer-valued fields **************** #
+# ******************************************************************* #
+
+# ------------------------------------------------------------------- #
+# ============== Adding integer fields ============================== #
+# ------------------------------------------------------------------- #
+
+class LFRicIntXPlusYKern(DynXPlusYKern):
```

Comment:
Once the naming is sorted out, these classes *could* all be generated from the corresponding real-data versions. That would save a lot of coding. What do you think?

Reply:
What would happen with the `__str__` method? If we're not fussed about "real" or "integer" in it, I'm happy with the generating integer built-ins from their real-data versions.